### PR TITLE
adds parameter info to the completion popup

### PR DESCRIPTION
**Pull request by kasperpeulen**
This is not perfect, when there are long names:

I tried different methods, but it kind of strechtes my head. I think I need to put the hint list in an extra div to get something that works properly.

### CODE REVIEW

File: lib/editing/editor_codemirror.css [hunk 1]
```diff
@@ -53,12 +53,15 @@
...
-  border-radius: 0px;
+  border-radius: 0;
+  max-width: 30em; /* overrides the default 19em in show-hint.css */
+  padding-right: 10px;
 }
 
 li.CodeMirror-hint-active {
   background: #545454;
   color: #f0dfaf;
+  max-width: none;
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.css]: Perhaps drop this.

Comment (kasperpeulen) [inline lib/editing/editor_codemirror.css]: I was thinking about that, but I guess some people want to see that info. What do you think of this solution:

**Comment by devoncarew** [comment]
So, my thoughts are:

- we want to avoid horizontal scrolling, esp. in a popup window
- at some point in the future, we may have a 'details' section to the right of the completion popup. That could show the full completion info, including dartdoc.
- for now, just indicating that we overflowed would work. So, some `...` when completion info is too long

File: lib/editing/editor_codemirror.css [hunk 2]
```diff
@@ -47,18 +47,36 @@
...
+  right: 0;
+  bottom: 0;
 }
 
 .CodeMirror-hint {
   background: #3f3f3f;
   color: #dcdccc;
-  border-radius: 0px;
+  border-radius: 0;
+  max-width: 30em; /* overrides the default 19em in show-hint.css */
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.css]: And maybe add this: `text-overflow: ellipsis;`?

File: lib/editing/editor_codemirror.css (hunk 1 above)
Comment (devoncarew) [inline lib/editing/editor_codemirror.css]: It's a little jarring that the box resizes. They _do_ get to see more of the completion info. I think on the whole having the popup stay the same size, even given the tradeoff, seems like a smoother UX.

**Comment by devoncarew** [comment]
@kasperpeulen, let's get @lukechurch's opinion on this as well.

**Comment by kasperpeulen** [comment]
Sure. What about showing the extra information on click ? @devoncarew  @lukechurch

**Comment by lukechurch** [comment]
Hey @kasperpeulen I think for now just ... ing out the extra information is probably the way to go. Having the box change size is a little jarring, and it's very non-standard UX to have clicking on an entry in a list have it expand the list rather than 'select the entry'

On the other hand I love the parameter display!

File: lib/completion.dart [hunk 3]
```diff
@@ -63,7 +63,7 @@ class DartCompleter extends CodeCompleter {
       List<Completion> completions =  analysisCompletions.map((completion) {
         // TODO: Move to using a LabelProvider; decouple the data and rendering.
         String displayString = completion.isMethod
-            ? '${completion.text}()' : completion.text;
+            ? completion.text + completion.parameters : completion.text;
```
Comment (lukechurch) [inline lib/completion.dart]: This is pretty frowned on by some in Dart land, could you make it "${completion.text} ${completion.parameters}"

**Comment by kasperpeulen** [comment]
@lukechurch It is not completely non-standard, intelliJ also works this way:


But okay, you guys decide :) I understand the downsides of my design as well. I will remove it.

**Comment by lukechurch** [comment]
@kasperpeulen Thanks! I'll merge

**Pull request merged**